apps/sms/views/shared/test/unit/contacts_test.js

```diff
@@ -29,406 +33,314 @@ suite('Contacts', function(done) {
-            filter.filterValue === notFoundNumber ||
-              filter.filterValue === notFoundEmail) {
-          result = [];
-        }
-
-        // Supports two "no match" cases
-        if (filter.filterValue === '911' ||
-            filter.filterValue === 'wontmatch' ||
-              filter.filterValue === 'z@y.com') {
-          result = [];
-        }
-
-        if (filter.filterValue === 'jane') {
-          result = MockContact.list([
-            // true
-            { givenName: ['Jane'], familyName: ['D'] },
-            // false
-            { givenName: ['jane'], familyName: ['austen'] },
-            // true
-            { givenName: ['jane'], familyName: ['doe'] },
-            // false
-            { givenName: ['jane'], familyName: ['fonda'] },
-            // true
-            { givenName: ['jane'], familyName: ['dow'] },
-            // false
-            { givenName: ['janet'], familyName: [''] }
-          ]);
-        }
-
-        if (filter.filterValue === 'julien') {
-          result = MockContact.list([
-            { givenName: ['Julien'] }
-          ]);
-        }
-
-        if (filter.filterValue === 'do') {
-          result = MockContact.list([
-            // true
-            { givenName: ['Jane'], familyName: ['Doozer'] },
-            // false
-            { givenName: ['doug'], familyName: ['dooley'] },
-            // true
-            { givenName: ['jane'], familyName: ['doe'] },
-            // false
-            { givenName: ['jerry'], familyName: ['doe'] },
-            // true
-            { givenName: ['j'], familyName: ['dow'] },
-            // true
-            { givenName: ['john'], familyName: ['doland'] }
-          ]);
-        }
-
-        if (filter.filterValue === 'doozer') {
-          result = MockContact.list([
-            { givenName: ['Jane'], familyName: ['Doozer'] }
-          ]);
-        }
-
-        if (filter.filterValue === 'mary') {
-          result = MockContact.list([
-            { givenName: ['Mary Anne'], familyName: ['Jones'] }
-          ]);
-        }
-
-        if (filter.filterValue === 'callonerror') {
-          result = null;
-        }
-
-        // All other cases
-
-        if (result === null) {
-          return Promise.reject(
-            new Error('Mock missing filter params')
-          );
-        }
-
-        return Promise.resolve(result);
-      }
-    };
-
     var mockReaderUtils = new MockFbReaderUtilsObj();
     mockReaderUtils.targetFbNumber = targetFbNumber;
     mockReaderUtils.fbContactName = fbContactName;
     window.fb = mockReaderUtils;
   });
 
-  teardown(function() {
-    navigator.mozContacts.mHistory.length = 0;
-  });
-
   suiteTeardown(function() {
     navigator.mozContacts = nativeMozContacts;
     window.fb = realFb;
   });
 
   suite('Contacts.findContactByString, single-term', function() {
+    setup(function(){
+      this.sinon.stub(navigator.mozContacts, 'find');
+    });
```

Comment:
You can put this setup in the top level suite -- and then you don't need to repeat it in every subsuite :)